device.go

```diff
@@ -128,8 +128,13 @@ func (d *Device) Config(cfgNum int) (*Config, error) {
 		dev:     d,
 		claimed: make(map[int]bool),
 	}
-	if err := libusb.setConfig(d.handle, uint8(cfgNum)); err != nil {
-		return nil, fmt.Errorf("failed to set active config %d for the device %s: %v", cfgNum, d, err)
+	
+	activeCfgNum, _ := d.ActiveConfigNum()
```

Comment:
don't ignore the error here, if an error is returned then the activeCfgNum cannot be trusted. If ActiveConfigNum returns 0+err, we should not attempt setConfig, but bail.
if c, err := d.ActiveConfigNum(); err != nil {
  return nil, fmt.Errorf("failed to query active config of the device %s: %v", d, err)
} else if cfgNum != c {
  if err := libusb.setConfig(...) {
    ...
  }
}